tests/phpunit/tests/test-rest-languages.php

```diff
@@ -0,0 +1,338 @@
+			$this->assertSame( $value, $data[ $name ] );
+		}
+	}
+
+	public function test_delete_language() {
+		wp_set_current_user( self::$administrator );
+		self::create_language( 'fr_FR' );
+
+		$request  = new WP_REST_Request( 'DELETE', '/pll/v2/languages/fr' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertSame( 200, $response->get_status() );
+
+		$data = $response->get_data();
+		$this->assertIsArray( $data );
+		$this->assertArrayHasKey( 'deleted', $data );
+		$this->assertTrue( $data['deleted'] );
+		$this->assertArrayHasKey( 'previous', $data );
+		$this->assertIsArray( $data['previous'] );
+
+		$this->assertFalse( $this->pll_env->model->get_language( 'fr_FR' ) );
+	}
+
+	/**
+	 * @testWith ["0", 401]
+	 *           ["author", 403]
+	 *
+	 * @param string $user
+	 * @param int    $status
+	 * @return void
+	 */
+	public function test_permissions( string $user, int $status ) {
+		if ( '0' === $user ) {
+			wp_set_current_user( 0 );
+		} else {
+			wp_set_current_user( self::$author );
+		}
+
+		self::create_language( 'fr_FR' );
+
+		// Delete language.
+		$request  = new WP_REST_Request( 'DELETE', '/pll/v2/languages/fr' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertSame( $status, $response->get_status() );
+		$this->assertInstanceOf( PLL_Language::class, $this->pll_env->model->get_language( 'fr_FR' ) );
+
+		// Update language.
+		$request = new WP_REST_Request( 'PATCH', '/pll/v2/languages/fr' );
+		$request->set_param( 'name', 'François' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertSame( $status, $response->get_status() );
+		$this->assertNotSame( 'François', $this->pll_env->model->get_language( 'fr_FR' )->name );
+
+		// Create language.
+		$request = new WP_REST_Request( 'POST', '/pll/v2/languages' );
+		$request->set_param( 'locale', 'es_ES' ); // Required.
+		$response = $this->server->dispatch( $request );
+
+		$this->assertSame( $status, $response->get_status() );
+		$this->assertFalse( $this->pll_env->model->get_language( 'es_ES' ) );
+	}
+
+	/**
+	 * @testWith ["PUT"]
+	 *           ["PATCH"]
+	 *           ["DELETE"]
+	 *
+	 * @param string $method
+	 * @return void
+	 */
+	public function test_missing_params( string $method ) {
+		wp_set_current_user( self::$administrator );
+		self::create_language( 'fr_FR' );
+
+		$request  = new WP_REST_Request( $method, '/pll/v2/languages' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertSame( 404, $response->get_status() );
+
+		$data = $response->get_data();
+		$this->assertIsArray( $data );
+		$this->assertArrayHasKey( 'code', $data );
+		$this->assertSame( 'rest_no_route', $data['code'] );
+	}
+
+	/**
+	 * @testWith ["PUT"]
+	 *           ["DELETE"]
+	 *
+	 * @param string $method
+	 * @return void
+	 */
+	public function test_invalid_param( string $method ) {
+		wp_set_current_user( self::$administrator );
+		self::create_language( 'fr_FR' );
```

Comment:
Please use the factory.

Reply:
Done in [898088f](https://github.com/polylang/polylang/pull/1505/commits/898088fc25c51c88bd330f763601683fcc3d8ee9).